plugins/couchbase/couchbaseembed.cpp

```diff
@@ -63,6 +63,31 @@ namespace couchbaseembed
     // Plugin Classes
     //--------------------------------------------------------------------------
 
+    void reportIfQueryFailure(Couchbase::Query * query)
+    {
+        auto status = query->meta().status();
+        if (status.errcode())
+        {
+            if (status.isNetworkError())
```

Comment:
Would it make sense to change all these conditionals to only set the error name string at the beginning of the string, then have a single call to failx ?